compiler/rustc_query_system/src/query/job.rs

```diff
@@ -232,7 +225,8 @@ impl QueryLatch {
         info.complete = true;
         let registry = rayon_core::Registry::current();
         for waiter in info.waiters.drain(..) {
-            waiter.notify(&registry);
+            rayon_core::mark_unblocked(&registry);
+            waiter.condvar.notify_one();
```

Comment:
Do we also need to unblock all the waiters first here,which helps reduce the chance of deadlocks due to work-stealing?

Reply:
No, this runs on an active Rayon thread, which prevents it from hitting the deadlock handler until all threads are unblocked.

Reply:
OK, it makes sense